### DISCUSSION TITLE
Custom link to go from rear port to front port

### DISCUSSION THREAD

**Discussion by pv2b** [netbox-community/netbox#13670] [Ideas]
Hello!

I thought I'd share a custom link that I've created for our environment. I find myself being on the page for a rear port and wanting to get to the corresponding front port. Surprisingly, it doesn't seem like there's a button for that in the UI. So I created one using a custom link.

Here is all you need to have a button just like this in your own setup:

1. Navigate to Customization > Custom Links, and click on the green "Add" button.
2. Paste and import this following blob of YAML:
```yaml
---
name: Link to front port from rear port
content_types: dcim.rearport
link_text: |
  {%- set fpcount = object.frontports.count() -%}
  {%- if fpcount == 1 -%}
        Show front port
  {%- elif fpcount > 1 -%}
        Show front ports
  {%- endif -%}
link_url: |
  {%- set fpcount = object.frontports.count() -%}
  {%- if fpcount == 1 -%}
        /dcim/front-ports/{{ object.frontports.get().id }}/
  {%- elif fpcount > 1 -%}
        /dcim/front-ports/?
  {%-   set sep = joiner("&") -%}
  {%-   for fp in object.frontports.all() -%}
          {{ sep() }}id={{ fp.id }}
  {%-   endfor -%}
  {%- endif -%}
weight: 100
button_class: blue
```

I hope this is useful for someone. :-)

**Comment by candlerb** [comment]
Nice work, thanks! You could contribute to the [customizations](https://github.com/netbox-community/customizations) repo, in a new custom_links directory.

I suppose the way to do this as a built-in solution would be:
* Add `rear_port_id` to FrontPortFilterSet
* Add a `Front Port(s)` field on the `/dcim/rear-ports/NN/` view which links to `/dcim/front-ports/?rear_port_id=NN` (unless positions=1, in which case it can link directly to a single front port)

**Reply by pv2b** [reply]
Thanks, I had no idea that repo existed. I might submit it there later when I have the time.

**Reply by pv2b** [reply]
Just a note on your proposal here, checking for positions=1 isn't the right way of doing it, because you might have a rear port that has no corresponding front-port, or fewer (or more! in case of db inconsistency) ports than positions. You probably want to actually count the associated front ports (like I'm doing) instead of relying on positions being accurate

**Reply by deku-m** [reply]
Thanks for the solution, its weird this isnt default like with front port
